### PR TITLE
Srwiseman 6272 fix

### LINKED ISSUE #6272: Zoom When Model Load Failure
When loading a CZML file and the model fails to load it causes the camera to never zoom to the entity.  This is most likely because there is no detection of the model load failure.  

Ideally, the model load failure would be detected and then the camera would zoom to the entity anyways.

### PR DESCRIPTION
**Pull request by srwiseman**
Fixes #6272.

This was a pretty basic fix, but it was causing me some issues since `model.readyPromise` evaluates within a loop. Therefore, if I tried to create a function like `.otherwise(function() {...});`,  eslint would fail with 'no-loop-func'.

To solve this, I created a function `checkModelLoad()`. If this function determines the the model will not load, it sets a flag called `loadFail` within the model data. This flag is then checked when the `boundingSphereState` is evaluated.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
[Please sign the CLA before we review this PR.](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla)

Welcome to the Cesium community @srwiseman!

Can you please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) so that we can review and merge this pull request?

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by pjcozzi** [comment]
Thanks @srwiseman.  We received your CLA.

@lilleyse can you review this when available?

File: Source/DataSources/ModelVisualizer.js [hunk 1]
```diff
@@ -327,6 +327,12 @@ define([
     function onModelError(error) {
         console.error(error);
     }
+    function checkModelLoad(model, entity, modelHash){
+        model.readyPromise.otherwise(function(error){
+            onModelError(error);
```
Comment (lilleyse) [inline Source/DataSources/ModelVisualizer.js]: It's fine now to just copy the contents of `onModelError` into here and remove that function.

File: Source/DataSources/ModelVisualizer.js [hunk 2]
```diff
@@ -137,9 +134,12 @@ define([
                     url : resource.url,
                     animationsRunning : false,
                     nodeTransformationsScratch : {},
-                    originalNodeMatrixHash : {}
+                    originalNodeMatrixHash : {},
+                    loadFail: false
```
Comment (lilleyse) [inline Source/DataSources/ModelVisualizer.js]: Small style tweak, add a space before the `:`

` loadFail : false`

**Comment by lilleyse** [comment]
Also add the fix to `CHANGES.md`.

**Comment by srwiseman** [comment]
Thanks! If I have time, I'll make these changes this evening.

**Comment by srwiseman** [comment]
@lilleyse Done!

**Comment by lilleyse** [comment]
Thanks @srwiseman.

**Comment by mramato** [comment]
@lilleyse is there a reason we merged this without tests?

**Comment by lilleyse** [comment]
Ah... my bad. Yes there should be a test. @srwiseman would you like to add one, or leave that to us?

**Comment by srwiseman** [comment]
Hi @lilleyse, sorry I was on vacation with no internet for a while.

Did you write a test for this yet? If not, I could write one.

**Comment by lilleyse** [comment]
Ah no problem. I haven't written a test, so you are still welcome.

**Comment by srwiseman** [comment]
Cool. So basically the test would check to ensure that if the model fails to load, the bounding sphere state correctly gets updated to FAILED? Correct?

**Comment by lilleyse** [comment]
Yes that sounds good.

**Pull request merged**